src/scripts/main.js

```diff
@@ -1,3 +1,81 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field');
+const rowButtons = document.querySelectorAll('[class*="row"]');
+const columnButtons = document.querySelectorAll('[class*="column"]');
+const rows = [...table.querySelectorAll('tr')];
+const column = rows[0].cells;
+let rowsNumber = rows.length;
+let columnNumber = column.length;
+
+document.body.addEventListener('click', (e) => {
+  const button = e.target.closest('.button');
+
+  if (button) {
+    const buttonFunc = [...button.classList];
+    const regex = /column/g;
+    const containColumns = buttonFunc.some((cls) => regex.test(cls));
+
+    table.innerHTML = '';
+
+    if (containColumns) {
+      const appendRegex = /append/g;
+      const isRegexTrue = buttonFunc.some((cls) => appendRegex.test(cls));
+
+      if (isRegexTrue) {
+        columnNumber++;
+
+        if (columnNumber >= 10) {
+          button.setAttribute('disabled', '');
+          columnNumber = 10;
+        } else {
+          columnButtons.forEach((but) => but.removeAttribute('disabled'));
+        }
+      } else {
+        columnNumber--;
+
+        if (columnNumber <= 2) {
```

Comment:
Setting the 'columnNumber' to 2 after disabling the button is redundant because the condition already ensures that 'columnNumber' is 2 or less. You can remove the assignment on line 39.